connection.go

```diff
@@ -430,10 +432,23 @@ func (mc *mysqlConn) getSystemVar(name string) ([]byte, error) {
 	return nil, err
 }
 
+const cancelConnectionStartTimeout = time.Second
+
 // finish is called when the query has canceled.
 func (mc *mysqlConn) cancel(err error) {
 	mc.canceled.Set(err)
 	mc.cleanup()
+	cancelContext, _ := context.WithTimeout(context.Background(), cancelConnectionStartTimeout)
+	connector := connector{mc.cfg}
+	cancelConn, err := connector.Connect(cancelContext)
```

Comment:
connector should cache the connection for cancellation.

Reply:
And sending `KILL CONNECTION` after `mc.cleanup()` is very dangerous.
The connection might be closed already, and the same connection id may be used by healthy new connection.

MySQL doesn't provide safe way to kill connection (that's one reason I don't want to make it default), but safer way is:

1. Send `KILL CONNECTION` from other connection.
2. Wait response (it might be successful result, or cancelled error).

Reply:
Done

Reply:
> connector should cache the connection for cancellation.

Done